### ISSUE TITLE
Newenvironment with "old" environment not working

### ISSUE THREAD

**Issue by zorkow** [mathjax/MathJax-src#1137]
Example:

```
\newenvironment{boxed}
    {
    \begin{array}{|c|c|}\hline
    }
    {
    \\\hline
    \end{array} 
    }
\begin{boxed}a&b\\c&d\end{boxed}
```

Worked fine in `3.2.2`, stopped working in `4.0.0-alpha.1`.

**Comment by dpvc** [comment]
I suspect 9ed1360c2c4cb1c89d71f5951248024ca062020a is the commit that is the culprit, but need to look further

**Comment by dpvc** [comment]
The issue that commit fixed is described in #856.

**Comment by dpvc** [comment]
> That's because we did not have a complete set of tests!

Yes, certainly.  I know you are working on that, and I will, too.

I have made a PR to fix the problem.  Sorry about the duplicate PRs, the first was opened as a draft, and I couldn't find where to change that, so resubmitted it.

**Comment by dpvc** [comment]
I added these sets into the `test-newcommand` branch.  If you had others, you can add a commit to that.  Closing this issue now.
